From 4b59d129fd1026bab37256af0df9ae7ace39e7ba Mon Sep 17 00:00:00 2001
From: Michael Tremer <michael.tremer@ipfire.org>
Date: Mon, 27 Apr 2015 18:49:45 +0200
Subject: [PATCH] stroke: Increase stroke buffer size to 8k

Complicated connections can have lots of arguments
for the ike= and esp= directives in the ipsec.conf
configuration file. strongSwan wouldn't import those
because the size of the message that is send from
stroke to charon exceeded the limit of 4k.

This patch increases the size of the buffer that
can be passed to charon to 8k which should be enough
even for connections with longer configurations.
---
 src/stroke/stroke_msg.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stroke/stroke_msg.h b/src/stroke/stroke_msg.h
index c2b923f6db9a..c391efa00105 100644
--- a/src/stroke/stroke_msg.h
+++ b/src/stroke/stroke_msg.h
@@ -32,7 +32,7 @@
  */
 #define STROKE_SOCKET IPSEC_PIDDIR "/charon.ctl"
 
-#define STROKE_BUF_LEN		4096
+#define STROKE_BUF_LEN		8192
 
 typedef enum list_flag_t list_flag_t;
 
-- 
2.1.0

